### ISSUE TITLE
Mailjet Integration cannot save form - List ID field is missing

### ISSUE THREAD

**Issue by Maybach91** [verbb/formie#1138]
### Describe the bug

If I want to integrate Mailjet with a form, i can successful add the integration itself and it says `connected`. When i want to add the integration to a form, i cannot save the whole form, because it says the following error in the JSON response:

```json

        "mailjet": {
          "enabled": "1",
          "optInField": "",
          "errors": {
            "listId": [
              "List Id cannot be blank."
            ]
          }
        }      

```
<details>
<summary>full json response</summary>

<code>
{
  "success": false,
  "id": 3423,
  "notifications": [],
  "errors": [],
  "fieldLayoutId": 54,
  "config": {
    "handle": "newsletterSubscribeForm",
    "oldHandle": "newsletterSubscribeForm",
    "fieldContentTable": "{{%fmc_newslettersubscribeform}}",
    "templateId": null,
    "submitActionEntryId": null,
    "submitActionEntrySiteId": null,
    "defaultStatusId": 1,
    "dataRetention": "forever",
    "dataRetentionValue": "",
    "userDeletedAction": "retain",
    "fileUploadsAction": "retain",
    "fieldLayoutId": 54,
    "settings": {
      "displayFormTitle": true,
      "displayCurrentPageTitle": false,
      "displayPageTabs": false,
      "displayPageProgress": false,
      "scrollToTop": true,
      "progressPosition": "end",
      "defaultLabelPosition": "verbb\\formie\\positions\\AboveInput",
      "defaultInstructionsPosition": "verbb\\formie\\positions\\AboveInput",
      "submitMethod": "ajax",
      "submitAction": "message",
      "submitActionTab": null,
      "submitActionUrl": null,
      "submitActionFormHide": false,
      "submitActionMessage": "[{\"type\":\"paragraph\",\"attrs\":{\"textAlign\":\"start\"},\"content\":[{\"type\":\"text\",\"text\":\"Submission saved.\"}]}]",
      "submitActionMessageTimeout": null,
      "submitActionMessagePosition": "top-form",
      "loadingIndicator": "spinner",
      "loadingIndicatorText": null,
      "validationOnSubmit": true,
      "validationOnFocus": false,
      "errorMessage": "[{\"type\":\"paragraph\",\"attrs\":{\"textAlign\":\"start\"},\"content\":[{\"type\":\"text\",\"text\":\"Couldn’t save submission due to errors.\"}]}]",
      "errorMessagePosition": "top-form",
      "requireUser": false,
      "requireUserMessage": "[{\"type\":\"paragraph\",\"attrs\":{\"textAlign\":\"start\"}}]",
      "scheduleForm": false,
      "scheduleFormStart": null,
      "scheduleFormEnd": null,
      "scheduleFormPendingMessage": "[{\"type\":\"paragraph\",\"attrs\":{\"textAlign\":\"start\"}}]",
      "scheduleFormExpiredMessage": "[{\"type\":\"paragraph\",\"attrs\":{\"textAlign\":\"start\"}}]",
      "limitSubmissions": false,
      "limitSubmissionsNumber": null,
      "limitSubmissionsType": "total",
      "limitSubmissionsMessage": "[{\"type\":\"paragraph\",\"attrs\":{\"textAlign\":\"start\"}}]",
      "integrations": {
        "recaptcha": {
          "enabled": "1",
          "showAllPages": ""
        },
        "hcaptcha": {
          "enabled": "1",
          "showAllPages": ""
        },
        "duplicate": {
          "enabled": "",
          "showAllPages": ""
        },
        "honeypot": {
          "enabled": "",
          "showAllPages": ""
        },
        "javascript": {
          "enabled": "",
          "showAllPages": ""
        },
        "mailjet": {
          "enabled": "1",
          "optInField": "",
          "errors": {
            "listId": [
              "List Id cannot be blank."
            ]
          }
        }
      },
      "submissionTitleFormat": "{timestamp}",
      "collectIp": false,
      "collectUser": false,
      "dataRetention": null,
      "dataRetentionValue": null,
      "fileUploadsAction": null,
      "redirectUrl": null,
      "defaultEmailTemplateId": null,
      "disableCaptchas": false
    },
    "resetClasses": false,
    "id": 3423,
    "tempId": null,
    "draftId": null,
    "revisionId": null,
    "isProvisionalDraft": false,
    "uid": "8e8586bb-bad4-4260-8469-fbd3efe20f9f",
    "siteSettingsId": 3877,
    "contentId": 821,
    "enabled": true,
    "archived": false,
    "siteId": 4,
    "title": "Newsletter Anmeldung Formular",
    "slug": null,
    "uri": null,
    "dateCreated": "2022-10-24T16:35:33+02:00",
    "dateUpdated": "2022-10-24T17:02:50+02:00",
    "dateLastMerged": null,
    "dateDeleted": null,
    "trashed": false,
    "isNewForSite": false,
    "canonicalId": 3423,
    "isDraft": false,
    "isRevision": false,
    "isUnpublishedDraft": false,
    "ref": null,
    "status": "enabled",
    "structureId": null,
    "url": null,
    "pages": [
      {
        "id": 3713,
        "label": "Page 1",
        "errors": [],
        "hasError": false,
        "rows": [
          {
            "id": null,
            "fields": [
              {
                "id": 97,
                "name": "Vorname",
                "handle": "vorname",
                "columnSuffix": "wqcwgkir",
                "label": "Vorname",
                "icon": "<svg aria-hidden=\"true\" focusable=\"false\" data-prefix=\"far\" data-icon=\"text-width\" class=\"svg-inline--fa fa-text-width fa-w-14\" role=\"img\" xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 448 512\"><path fill=\"currentColor\" d=\"M0 139.131V44c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v95.131c0 6.627-5.373 12-12 12h-26.747c-6.627 0-12-5.373-12-12V80H251.238v192H300c6.627 0 12 5.373 12 12v24c0 6.627-5.373 12-12 12H148c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h48.762V80H50.125v59.131c0 6.627-5.373 12-12 12H12c-6.627 0-12-5.372-12-12zm444.241 252.145l-72.001-67.994c-7.581-7.16-20.24-1.863-20.24 8.724V376H96v-43.993c0-10.58-12.652-15.89-20.24-8.724L3.759 391.276c-5.014 4.735-5.01 12.716 0 17.448l72.001 67.994c7.581 7.16 20.24 1.863 20.24-8.724V424h256v43.993c0 10.58 12.652 15.89 20.24 8.724l72.001-67.993c5.014-4.735 5.01-12.716 0-17.448z\"></path></svg>",
                "type": "verbb\\formie\\fields\\formfields\\SingleLineText",
                "errors": [],
                "hasLabel": true,
                "hasError": false,
                "settings": {
                  "limit": false,
                  "min": null,
                  "minType": "characters",
                  "max": null,
                  "maxType": "characters",
                  "matchField": null,
                  "placeholder": "z.B. Maxi",
                  "defaultValue": null,
                  "prePopulate": null,
                  "errorMessage": null,
                  "labelPosition": null,
                  "instructionsPosition": null,
                  "cssClasses": null,
                  "containerAttributes": null,
                  "inputAttributes": null,
                  "includeInEmail": true,
                  "enableConditions": false,
                  "conditions": null,
                  "enableContentEncryption": false,
                  "visibility": null,
                  "formId": 3423,
                  "isNested": false,
                  "label": "Vorname",
                  "handle": "vorname",
                  "required": false,
                  "instructions": null
                },
                "isCosmetic": false,
                "isSynced": false,
                "hasConditions": false,
                "supportsNested": false,
                "isNested": false
              }
            ]
          },
          {
            "id": null,
            "fields": [
              {
                "id": 98,
                "name": "Nachname",
                "handle": "nachname",
                "columnSuffix": "tlsvlwbj",
                "label": "Nachname",
                "icon": "<svg aria-hidden=\"true\" focusable=\"false\" data-prefix=\"far\" data-icon=\"text-width\" class=\"svg-inline--fa fa-text-width fa-w-14\" role=\"img\" xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 448 512\"><path fill=\"currentColor\" d=\"M0 139.131V44c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v95.131c0 6.627-5.373 12-12 12h-26.747c-6.627 0-12-5.373-12-12V80H251.238v192H300c6.627 0 12 5.373 12 12v24c0 6.627-5.373 12-12 12H148c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h48.762V80H50.125v59.131c0 6.627-5.373 12-12 12H12c-6.627 0-12-5.372-12-12zm444.241 252.145l-72.001-67.994c-7.581-7.16-20.24-1.863-20.24 8.724V376H96v-43.993c0-10.58-12.652-15.89-20.24-8.724L3.759 391.276c-5.014 4.735-5.01 12.716 0 17.448l72.001 67.994c7.581 7.16 20.24 1.863 20.24-8.724V424h256v43.993c0 10.58 12.652 15.89 20.24 8.724l72.001-67.993c5.014-4.735 5.01-12.716 0-17.448z\"></path></svg>",
                "type": "verbb\\formie\\fields\\formfields\\SingleLineText",
                "errors": [],
                "hasLabel": true,
                "hasError": false,
                "settings": {
                  "limit": false,
                  "min": null,
                  "minType": "characters",
                  "max": null,
                  "maxType": "characters",
                  "matchField": null,
                  "placeholder": "z.B. Musterwesen",
                  "defaultValue": null,
                  "prePopulate": null,
                  "errorMessage": null,
                  "labelPosition": null,
                  "instructionsPosition": null,
                  "cssClasses": null,
                  "containerAttributes": null,
                  "inputAttributes": null,
                  "includeInEmail": true,
                  "enableConditions": false,
                  "conditions": null,
                  "enableContentEncryption": false,
                  "visibility": null,
                  "formId": 3423,
                  "isNested": false,
                  "label": "Nachname",
                  "handle": "nachname",
                  "required": false,
                  "instructions": null
                },
                "isCosmetic": false,
                "isSynced": false,
                "hasConditions": false,
                "supportsNested": false,
                "isNested": false
              }
            ]
          },
          {
            "id": null,
            "fields": [
              {
                "id": 99,
                "name": "E-Mail",
                "handle": "eMail",
                "columnSuffix": "zkflpwjy",
                "label": "E-Mail",
                "icon": "<svg aria-hidden=\"true\" focusable=\"false\" data-prefix=\"far\" data-icon=\"envelope\" class=\"svg-inline--fa fa-envelope fa-w-16\" role=\"img\" xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 512 512\"><path fill=\"currentColor\" d=\"M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z\"></path></svg>",
                "type": "verbb\\formie\\fields\\formfields\\Email",
                "errors": [],
                "hasLabel": true,
                "hasError": false,
                "settings": {
                  "validateDomain": false,
                  "blockedDomains": [],
                  "uniqueValue": true,
                  "matchField": null,
                  "placeholder": "z.B. maxi.musterwesen@gmail.com",
                  "defaultValue": null,
                  "prePopulate": null,
                  "errorMessage": null,
                  "labelPosition": null,
                  "instructionsPosition": null,
                  "cssClasses": null,
                  "containerAttributes": null,
                  "inputAttributes": null,
                  "includeInEmail": true,
                  "enableConditions": false,
                  "conditions": null,
                  "enableContentEncryption": false,
                  "visibility": null,
                  "formId": 3423,
                  "isNested": false,
                  "label": "E-Mail",
                  "handle": "eMail",
                  "required": true,
                  "instructions": null
                },
                "isCosmetic": false,
                "isSynced": false,
                "hasConditions": false,
                "supportsNested": false,
                "isNested": false
              }
            ]
          },
          {
            "id": null,
            "fields": [
              {
                "id": 100,
                "name": "Datenschutz",
                "handle": "datenschutz",
                "columnSuffix": "djryutba",
                "label": "Datenschutz",
                "icon": "<svg aria-hidden=\"true\" focusable=\"false\" data-prefix=\"far\" data-icon=\"check-square\" class=\"svg-inline--fa fa-check-square fa-w-14\" role=\"img\" xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 448 512\"><path fill=\"currentColor\" d=\"M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352zm-35.864-241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z\"></path></svg>\n",
                "type": "verbb\\formie\\fields\\formfields\\Agree",
                "errors": [],
                "hasLabel": true,
                "hasError": false,
                "settings": {
                  "description": [
                    {
                      "type": "paragraph",
                      "attrs": {
                        "textAlign": "start"
                      },
                      "content": [
                        {
                          "type": "text",
                          "text": "Yes ich erlaube alles."
                        }
                      ]
                    }
                  ],
                  "checkedValue": "Yes",
                  "uncheckedValue": "No",
                  "matchField": null,
                  "placeholder": null,
                  "defaultValue": false,
                  "prePopulate": null,
                  "errorMessage": null,
                  "labelPosition": "verbb\\formie\\positions\\Hidden",
                  "instructionsPosition": null,
                  "cssClasses": null,
                  "containerAttributes": null,
                  "inputAttributes": null,
                  "includeInEmail": true,
                  "enableConditions": false,
                  "conditions": null,
                  "enableContentEncryption": false,
                  "visibility": null,
                  "formId": 3423,
                  "isNested": false,
                  "label": "Datenschutz",
                  "handle": "datenschutz",
                  "required": true,
                  "instructions": null
                },
                "isCosmetic": false,
                "isSynced": false,
                "hasConditions": false,
                "supportsNested": false,
                "isNested": false
              }
            ]
          }
        ],
        "settings": {
          "submitButtonLabel": "Submit",
          "backButtonLabel": "Back",
          "showBackButton": false,
          "saveButtonLabel": "Save",
          "showSaveButton": false,
          "saveButtonStyle": "link",
          "buttonsPosition": "left",
          "cssClasses": null,
          "containerAttributes": null,
          "inputAttributes": null,
          "enableNextButtonConditions": false,
          "nextButtonConditions": [],
          "enablePageConditions": false,
          "pageConditions": [],
          "enableJsEvents": false,
          "jsGtmEventOptions": []
        }
      }
    ],
    "errors": [],
    "hasError": false
  }
}
</code>

</details>

But there is no field to select a (subscriber/contact) list:
<img width="675" alt="image" src="https://user-images.githubusercontent.com/1867543/197565481-2358fcd8-8159-429a-84b3-9c87ce58f0db.png">
I’m using a fresh created Mailjet Account with contact lists there:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/1867543/197566655-1643adf8-b43e-4b26-9480-33f8b97b5688.png">

/cc @jmauzyk @engram-design  #1017 

# EDIT

When changing vendor/verbb/formie/src/templates/integrations/email-marketing/mailjet/_form-settings.html:56

from:
`:rows="getSourceFields('lists')"`

to:
`:rows="get(settings,'lists')"`

i can see this:
<img width="1638" alt="image" src="https://user-images.githubusercontent.com/1867543/197570935-95f037b1-8334-49db-a192-3e4de327e39c.png">

but refresh button is not working and i cannot see any of my lists. it looks like the integration has to be updated (sendinblue as well?)

### Steps to reproduce

1. Install latest formie plugin
2. Add `Mailjet` Integration
3. Create a form
4. Add `Mailjet` integration to form
5. Save

### Form settings

- Multi-page form: No
- Submission Method: Ajax AND Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.2.8

### Plugin version

2.0.13

### Multi-site?

Yes

### Additional context

I’m trying to save this on a local machine using valet.

**Comment by Maybach91** [comment]
Okay, i got the fix. The following wrapper is missing in the `_form-settings.html` template:
```vue
    <template v-slot="{ get, isEmpty, input, settings, sourceId, loading, refresh, error, errorMessage, getSourceFields }">
```

**Comment by engram-design** [comment]
Thanks for your PR, just cleaned up some formatting FYI. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.13",
  "...": "..."
}
```

Then run `composer update`.

**Comment by Maybach91** [comment]
It looks like the integration is still not working properly, since Mailjet doesn’t receive any data from formie. :(

**Comment by engram-design** [comment]
Can you check your `storage/logs/formie.log` file to give us more information on the error? I've just tested it on my end, and seems to work without issue.

**Comment by Maybach91** [comment]
I cannot get any error, the log looks good to me. Do you have any idea, how I can debug the integration? What is sending when to Mailjet, @engram-design?

I also have the issue, that the page is reloaded although i set ajax reloading. But i don’t get any javascript errors. Also using the `renderFormJs`:

formie.twig:
```twig
    {% set form = craft.formie.forms.handle(props.data.formHandle.one().handle).one() %}
    <form class="js-formie {{ props.data.name }} {{ props.classes.root }} {{ props.classes.custom }}"
          data-config="{{ form.configJson }}" id="{{ form.formId }}" method="post">
        {{ craft.formie.renderForm(form) }}

    </form>
    {{ craft.formie.renderFormJs(form) }}
    <script type="text/javascript">
      window.csrfTokenName = "{{ craft.app.config.general.csrfTokenName|e('js') }}";
      window.csrfTokenValue = "{{ craft.app.request.csrfToken|e('js') }}";
    </script>
```
  
formie.log:
```log
2022-10-27 16:47:53 [INFO] Submission triggered for newsletterSubscribeForm.
2022-10-27 16:47:53 [INFO] Request context:
$_GET = [
    'x-craft-live-preview' => '9F5CXMOEiB'
]

$_POST = [
    'CRAFT_CSRF_TOKEN' => '••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••'
    'action' => 'formie/submissions/submit'
    'submitAction' => 'submit'
    'handle' => 'newsletterSubscribeForm'
    'siteId' => '4'
    'fields' => [
        'vorname' => 'Patrick6'
        'nachname' => 'Ludewig6'
        'eMail' => 'validemail+34dsf324@gmail.com'
        'datenschutz' => '1'
    ]
]

```

**Comment by engram-design** [comment]
Are you using the queue to process integrations (the default for Formie)? Are there any failed queue jobs? The integration is only run when you visit the control panel, unless you configure it [otherwise](https://verbb.io/craft-plugins/formie/docs/get-started/troubleshooting).

I see you're also using live preview, so would be good to confirm it's working outside of that.

As for Ajax submissions not working on the front-end, it's because your markup is invalid. `renderForm()` will produce a `<form>` element itself, but you're wrapping another `<form>` element around this call, having nested `<form>` elements is invalid HTML.

And just by the way, I'm going to assume `props.data.formHandle` is a Formie form field. You don't need to query for the form twice, as the field will return a Form element, ready to go. You could do:

```twig
{% set form = props.data.formHandle.one() %}
```

**Comment by engram-design** [comment]
Fixed in [2.0.14](https://github.com/verbb/formie/releases/tag/2.0.14)
